gitclone/git_test.go

```diff
@@ -96,3 +100,151 @@ func Test_parseListBranchesOutput(t *testing.T) {
 		})
 	}
 }
+
+func Test_handleCheckoutError(t *testing.T) {
+	type args struct {
+		callback func() (map[string][]string, error)
+		tag      string
+		err      error
+		shortMsg string
+		branch   string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *step.Error
+	}{
+		{
+			name: "handleCheckoutError: generic error without branch recommendation",
+			args: args{
+				callback: func() (map[string][]string, error) { return nil, nil },
+				tag:      "fetch_failed",
+				err:      errors.New("Something bad happened"),
```

Comment:
Can it happen that there is a generic error (not branch error), bit we still manage to collect available branches?

Reply:
Technically yes, I was thinking about this scenario as well...